File: src/main/java/toby/jpa/service/IBrotherService.java
package toby.jpa.service;

import toby.jpa.dto.BrotherDto;

import java.util.Optional;

public interface IBrotherService  {
    Iterable<BrotherDto> listBrothers();
    BrotherDto createNewBrother(BrotherDto brotherDto);
    Optional<BrotherDto> getBrotherById(Long discordId);
    BrotherDto updateBrother(BrotherDto brotherDto);
    void deleteBrother(BrotherDto brotherDto);
    void deleteBrotherById(Long discordId);
    void clearCache();

}

File: src/main/java/toby/jpa/service/IBrotherService.kt
package toby.jpa.service

import toby.jpa.dto.BrotherDto

interface IBrotherService {
    fun listBrothers(): Iterable<BrotherDto?>?
    fun createNewBrother(brotherDto: BrotherDto): BrotherDto?
    fun getBrotherById(discordId: Long?): BrotherDto?
    fun updateBrother(brotherDto: BrotherDto?): BrotherDto?
    fun deleteBrother(brotherDto: BrotherDto?)
    fun deleteBrotherById(discordId: Long?)
    fun clearCache()
}
